tests/index_test.py

```diff
@@ -0,0 +1,218 @@
+                'login': 'reviewerName1'
+            },
+            'created_at': get_past_time(hours=23)
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName2'
+            },
+            'created_at': get_past_time(hours=19)
+        }]
+        self.test_template = "{{ username }}\n{{ pr_list }}"
+
+    def mock_all_get_requests(self, mock_request: requests_mock) -> None:
+        param_page_1='?page=1&per_page=100'
+        param_page_2='?page=2&per_page=100'
+        mock_request.get(
+            github_services.PULL_REQUESTS_URL_TEMPLATE.format(
+                self.orgName, self.repoName) + param_page_1,
+            text=json.dumps(self.pull_response))
+        mock_request.get(
+            github_services.PULL_REQUESTS_URL_TEMPLATE.format(
+                self.orgName, self.repoName) + param_page_2,
+            text=json.dumps([]))
+
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 123) + param_page_1,
+            text=json.dumps(self.timeline1))
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 123) + param_page_2,
+            text=json.dumps([]))
+
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 234) + param_page_1,
+            text=json.dumps(self.timeline2))
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.orgName, self.repoName, 234) + param_page_2,
+            text=json.dumps([]))
+
+    def test_executing_main_function_sends_notification(self) -> None:
+        with requests_mock.Mocker() as mock_request:
+            self.mock_all_get_requests(mock_request)
+                
+            mock_resp_1 = Mock()
+            mock_resp_1.json.return_value = self.response_for_discussions
+
+            mock_resp_2 = Mock()
+            mock_resp_2.json.return_value = self.response_for_comment
+            with patch("requests.post", side_effect=[
+                mock_resp_1 if i % 2 == 0 else mock_resp_2 for i in range(6)]) as mock_post:
+
+                request_1 = requests.post(github_services.GITHUB_GRAPHQL_URL)
+                request_2 = requests.post(github_services.GITHUB_GRAPHQL_URL)
+                file_data = mock_open(read_data=self.test_template)
+                with patch("builtins.open", file_data):
+                    index.main([
+                        '--repo', 'orgName/repo',
+                        '--category', 'test_category_name_1',
+                        '--title', 'test_discussion_title_1',
+                        '--max-wait-hours', '20',
+                        '--token', 'githubTokenForApiRequest'
+                    ])
+        self.assertTrue(request_1, self.response_for_discussions)
```

Comment:
Not sure what you are asserting here -- assertTrue is generally only used for booleans, but this isn't one? Do you mean assertEqual?

Reply:
You are right. I wanted to use assertEqual, updated this time